packages/calling-stateful-client/src/TeamsCallAgentDeclarative.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* @conditional-compile-remove(teams-call) */
+import { TeamsCallAgent as TeamsCallAgentBeta } from '@azure/communication-calling';
+import { ProxyCallAgentCommon } from './CallAgentDeclarativeCommon';
+import { CallContext } from './CallContext';
+/* @conditional-compile-remove(teams-call) */
+import { DeclarativeIncomingCall } from './IncomingCallDeclarative';
+import { InternalCallContext } from './InternalCallContext';
+import { disposeAllViews } from './StreamUtils';
+
+type TeamsCallAgent = never | /* @conditional-compile-remove(teams-call) */ TeamsCallAgentBeta;
+
+/**
+ * @public
+ * `DeclarativeTeamsCallAgent` extends and proxies the {@link @azure/communication-calling#TeamsCallAgent}
+ */
+export type DeclarativeTeamsCallAgent = TeamsCallAgent & /* @conditional-compile-remove(one-to-n-calling) */ {
+  /**
+   * @beta
+   * A readonly array that returns all the active `incomingCalls`.
+   * An active incoming call is a call that has not been answered, declined or disconnected.
+   *
+   * @Remark This attribute doesn't exist on the {@link @azure/communication-calling#CallAgent} interface.
+   * @returns readonly array of {@link DeclarativeIncomingCall}
+   */
+  incomingCalls: ReadonlyArray<DeclarativeIncomingCall>;
```

Comment:
This should be moved to `DeclarativeCallAgentCommon`

Reply:
We don't expose DeclarativeCallAgentCommon as a public type, so we still need it for  DeclarativeCallAgentand  DeclarativeTeamsCallAgent, but I've extract IncomingCallManagement, as a shared type between these two